client/actions/events/api.js

```diff
@@ -0,0 +1,361 @@
+                    const range = {}
+
+                    if (advancedSearch.dates.start) {
+                        range['dates.start'] = { gte: advancedSearch.dates.start }
+                    }
+
+                    if (advancedSearch.dates.end) {
+                        range['dates.end'] = { lte: advancedSearch.dates.end }
+                    }
+
+                    filter.range = range
+                },
+            },
+        // loop over actions and performs if conditions are met
+        ].forEach((action) => {
+            if (action.condition()) {
+                action.do()
+            }
+        })
+
+        // default filter
+        if (isEqual(filter, {}) && isEqual(must, []) && isEqual(mustNot, [])) {
+            filter.range = { 'dates.end': { gte: 'now/d' } }
+        }
+
+        switch (state) {
+            case ITEM_STATE.SPIKED:
+                must.push({ term: { state: ITEM_STATE.SPIKED } })
+                break
+            case ITEM_STATE.ALL:
+                break
+            case ITEM_STATE.ACTIVE:
+            default:
+                mustNot.push({ term: { state: ITEM_STATE.SPIKED } })
+        }
+
+        query.bool = {
+            must,
+            must_not: mustNot,
+        }
+
+        // Query the API and sort by date
+        return api('events').query({
+            page: page,
+            sort: '[("dates.start",1)]',
+            embedded: { files: 1 },
+            source: JSON.stringify({
+                query,
+                filter,
+            }),
+        })
+        // convert dates to moment objects
+        .then((data) => ({
+            ...data,
+            _items: data._items.map((item) => ({
+                ...item,
+                dates: {
+                    ...item.dates,
+                    start: moment(item.dates.start),
+                    end: moment(item.dates.end),
+                },
+            })),
+        }))
+    }
+)
+
+/**
+ * Action Dispatcher to re-fetch the current list of events
+ * It achieves this by performing a fetch using the params from
+ * the store value `events.lastRequestParams`
+ */
+const refetchEvents = () => (
+    (dispatch, getState) => {
+        const prevParams = selectors.getPreviousEventRequestParams(getState())
+
+        const promises = []
+        for (let i = 1; i <= prevParams.page; i++) {
+            const params = {
+                ...prevParams,
+                page: i,
+            }
+            dispatch({
+                type: EVENTS.ACTIONS.REQUEST_EVENTS,
+                payload: params,
+            })
+            promises.push(dispatch(self.query(params)))
+        }
+
+        return Promise.all(promises)
+        .then((responses) => {
+            let events = responses
+                .map((e) => e._items)
+                .reduce((a, b) => a.concat(b))
+
+            dispatch(self.receiveEvents(events))
+            dispatch(self.setEventsList(events.map((e) => e._id)))
```

Comment:
In this api module, I think we should avoid to call UI actions.
I would call api actions from a UI action, and in a `then() `, dispatch the following UI actions there